docs/proposals/workflow.md

```diff
@@ -0,0 +1,155 @@
+The latest release of this document can be found
+[here](http://releases.k8s.io/release-1.1/docs/proposals/workflow.md).
+
+Documentation for other releases can be found at
+[releases.k8s.io](http://releases.k8s.io).
+</strong>
+--
+
+<!-- END STRIP_FOR_RELEASE -->
+
+<!-- END MUNGE: UNVERSIONED_WARNING -->
+
+## Abstract
+
+A proposal to modify [`Job` resource](../../docs/user-guide/jobs.md) to implement a minimal
+[Workflow managment system](https://en.wikipedia.org/wiki/Workflow_management_system) in kubernetes.
+Workflows (aka DAG workflows since jobs are organized in a Direct Acyclic Graph) are ubiquitous
+in modern [job schedulers](https://en.wikipedia.org/wiki/Job_scheduler), see for example:
+
+* [luigi](https://github.com/spotify/luigi)
+* [ozie](http://oozie.apache.org/)
+* [azkaban](https://azkaban.github.io/)
+
+Most of the [job schedulers](https://en.wikipedia.org/wiki/List_of_job_scheduler_software) offer
+workflow functionality to some extent.
+
+## Use Cases
+
+* As a user I want to be able to define job chains such that the completion of a given job will
+trigger other jobs.
+
+## Implementation
+
+The idea of this proposal is to add a label selector for identifying "parent" jobs of the current
+Job API object. A "parent" job is one that the current job depends upon, and that must be
+completed before the current job can be scheduled. The strongest point in favor of this approach
+is the ability to re-use the current job implementation. Using a label selector the current
+job will simply become a graph of jobs: all the features/controllers that will be built on top
+of job (like for example [#11980](https://github.com/kubernetes/kubernetes/issues/11980)
+and [#17242](https://github.com/kubernetes/kubernetes/issues/17242) will be automatically
+extended to support workflows.
+A similar approach is implemented in Chronos for
+[dependent jobs](https://mesos.github.io/chronos/docs/api.html#adding-a-dependent-job).
+
+### API Object
+
+To implement _workflows_ the `Job` API should be modified:
+
+```go
+// JobSpec describes how the job execution will look like.
+
+// These are valid conditions of a job.
+const (
+	// JobComplete means the job has completed its execution.
+	JobComplete JobConditionType = "Complete"
+	// JobWaiting means the job is waiting for its parents to finish their tasks.
+	JobWaiting  JobConditionType = "Waiting"
+)
+
+type JobSpec struct {
+	...
+	// Job labels selector to detect parent jobs.
+	ParentSelector map[string]string `json:"parentSelector"`
+}
+
+```
+
+#### JobSpec
+
+A new label selector will be added to `job.spec`. The `job.spec.parentSelector` is a label query
+over a set of jobs.
+If all selected jobs are completed the `job` will be started immediately.
+If `job.spec.parentSelector` is absent the `job` will be started immediately.
+If `job.spec.parentSelector` is pointing to a non-existing job, the job will wait indefinitely or
+until a matching
+job will be created and completed accordingly
```

Comment:
GC of terminated jobs is a good point: somehow we should prevent _parent_ jobs from being deleted and I admit that this is a weak point of the proposal. An even weaker point is (cascade cancelation): since the dependency is `child to parent` there's no easy way to delete the started job (bad UX).